jsoniter-scala-examples/src/main/scala/com/github/plokhotnyuk/jsoniter_scala/examples/Example01.scala

```diff
@@ -36,10 +36,16 @@ object Example01 {
           StringLeaf(in.readString(null))
         case _ =>
           in.rollbackToken()
-          val d = in.readDouble() // use in.readBigDecimal(null) for exact precision
-          val i = d.toInt
-          if (d == i) IntLeaf(i)
-          else FloatLeaf(d.toFloat) // possible 1 ULP rounding error here
+          in.setMark()
+          var b = 0
```

Comment:
It will end up with `implicit numeric widening` in the next line while calling `in.nextByte()`

If I switch to one of `byte` declarations like below, it works fine.

a) `var b = 0.toByte`
or
b) `var b = 0.byteValue`

Maybe the option `a)` is better, since option `b` returns a primitive value.

Reply:
I would prefer `var b = 0: Byte`.